feat(across-v2): make tokens come from json

Gamaranto opened the pull request:
This PR makes the tokens used in the dapp come from a json file.
The important change here is that the `logoURI` must now point to a file inside the `public` folder, not in the `assets` folder.

Also, I've tried using [superstruct](https://docs.superstructjs.org/) for adding schema validation at runtime, but failed to make it work with typescript. I've left my attempt as commented code, in case anyone can make it work and play nice with TS.

Signed-off-by: Gamaranto <francesco@umaproject.org>

Tulun commented on src/data/tokens.json:
```diff
@@ -0,0 +1,282 @@
+      "symbol": "WETH",
+      "decimals": 18,
+      "logoURI": "/logos/weth-logo.svg",
+      "bridgePool": "0x7355Efc63Ae731f584380a9838292c7046c1e433"
```
Yeah i guess the answer is that we need an optional bridgePool *but* it needs to know it needs to exist on certain other objects? is that possible?

daywiss replied:
i think we need a separate structure for the bridgepool, which is technically now the "hubpool".  It should only exist once per instance of the app, but it may change from mainnet to a testnet depending on how we initialize the app

mrice32 commented on src/utils/constants.ts:
```diff
@@ -1,17 +1,15 @@
+import bobaLogo from "assets/boba-logo.svg";
 import { getAddress } from "./address";
+import rawTokens from "../data/tokens.json";
+//import * as superstruct from 'superstruct';
```
Is superstruct intended to be the validation layer?

daywiss replied:
from my testing it works great to validate data and work with tyepscript to assert that the data has the properties you expect using `superstruct.assert`.  Francesco ran into some issues which didnt allow validation of enum keys on a `record`, so you cant actually validate this entire json file, but you can validate the elements within it

mrice32 reviewed (commented):
Looks awesome! Agree with @daywiss that I think this will work really well once superstruct validation is up and running

Gamaranto commented:
Addressed all PR comments, to remove `bridgePool` from tokens I had to break the pool tab, but that is ok since it will be fixed by @daywiss PR shortly. 
The LP fee calculation has also been broken, but it should not matter for now either

Gamaranto commented on src/utils/constants.ts:
```diff
@@ -98,289 +96,21 @@ type TokenList = [
+function processRawTokens(tokens: typeof rawTokens): Record<ChainId, TokenList> {
+  return Object.entries(tokens)
+    .reduce((record, [rawChainId, rawTokens]) => {
+      const chainId: ChainId = Number(rawChainId);
```
added it , thanks for catching this

daywiss reviewed (approved):
nice

mrice32 reviewed (approved):
LGTM!

Gamaranto commented on src/utils/constants.ts:
```diff
@@ -94,293 +91,37 @@ type TokenList = [
+    }, {} as Record<ChainId, TokenList>);
+}
+export const TOKENS_LIST: Record<ChainId, TokenList> = processRawTokens(rawTokens);
+//superstruct.assert(TOKENS_LIST, TokensList);
```
removed thanks
